### DISCUSSION TITLE
Promp displays scrambled characters after init'ing

### DISCUSSION THREAD

**Discussion by dbwtx** [JanDeDobbeleer/oh-my-posh#5392] [Troubleshoot] [answered]
I have followed the installation docs with the exception of using a different font (Lilex Nerd Font) instead.  When I run the command `oh-my-posh init pwsh | Invoke-Expression` I get scrambled characters like the following screen shot:
![image](https://github.com/user-attachments/assets/1de4d773-62ad-42ab-84e3-b9ead722b226)

I have configured my settings to use the font, but I haven't even set a theme yet, so I don't think it has anything to do with the font, as it looks basically the same in other fonts.  I've searched around, but can only find issues related to the font, like where boxes display instead of rendering characters etc... This looks like it is pulling the wrong symbols somehow?

**Comment by lewis-yeung** [comment, marked as answer]
@dbwtx See [FAQ][faq] in the documentation. There is an encoding issue in PowerShell. As a workaround, you can add the following code to your PowerShell profile, before the `oh-my-posh init` command:

```powershell
[Console]::OutputEncoding = [Text.Encoding]::UTF8
```

[faq]: https://ohmyposh.dev/docs/faq#my-prompt-is-broken-after-upgrading-to-powershell-74

**Reply by dbwtx** [reply]
Thanks, I did see this, and didn't try it.  I was under the impression (based on this windows dosumentation: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_character_encoding?view=powershell-7.4) that Powershell 7 used UTF8 by default, and thought it must be a different issue. Perhaps it has something to do with the BOM vs no BOM versions of UTF8? I should have just given it a try I guess!  Thank you!

**Reply by JanDeDobbeleer** [reply]
@dbwtx unfortunately pwsh still doesn't default to UTF8. One day. I'll probably be retired by then 😅

**Reply by dbwtx** [reply]
I guess I saw this line "PowerShell (v6 and higher) defaults to utf8NoBOM for all text output" and in my mind I was like "thats still UTF8, so its probably fine" 🤦‍♂️.  Oh well

**Reply by JanDeDobbeleer** [reply]
@dbwtx it's fair assumption to make. Encoding is a whole expertise on its own.

**Reply by madrum** [reply]
This was driving me crazy! Thank you!!!

**Reply by dbwtx** [reply]
This issue has been reopened by the Powershell team over at MS to at least investigate.  Hopefully that leads to a legit fix. Just though people might want to know.
